2-WebApp-graph-user/2-5-HybridFlow/README.md

```diff
@@ -0,0 +1,480 @@
+
+   > In the steps below, "ClientID" is the same as "Application ID" or "AppId".
+
+  1. Open the `appsettings.json` file.
+  1. Find the key `Domain` and replace the existing value with your Azure AD tenant name.
+  1. Find the key `TenantId` and replace the existing value with your Azure AD tenant ID.
+  1. Find the key `ClientId` and replace the existing value with the application ID (clientId) of `HybridFlow-aspnetcore` app copied from the Azure portal.
+  1. Find the key `ClientSecret` and replace the existing value with the **client secret** of `HybridFlow-aspnetcore` app copied from the Azure portal.
+</details>
+
+### Step 4: Running the sample
+
+ For command line run the next commands:
+
+```console
+    cd 2-WebApp-graph-user\2-5-HybridFlow
+    dotnet run
+```
+
+## Troubleshooting
+
+<details>
+ <summary>Expand for troubleshooting info</summary>
+
+Use [Stack Overflow](http://stackoverflow.com/questions/tagged/msal) to get support from the community.
+Ask your questions on Stack Overflow first and browse existing issues to see if someone has asked your question before.
+Make sure that your questions or comments are tagged with [`azure-active-directory` `adal` `msal` `dotnet`].
+
+If you find a bug in the sample, please raise the issue on [GitHub Issues](../../issues).
+
+To provide a recommendation, visit the following [User Voice page](https://feedback.azure.com/forums/169401-azure-active-directory).
+</details>
+
+## Using the sample
+
+<details>
+ <summary>Expand to see how to use the sample</summary>
+ Open your web browser and navigate to `https://localhost:7089`. The app immediately attempts to authenticate you via the Microsoft identity platform endpoint. Sign in using an user account in that tenant. The client side `MSAL.js` client will receive an **auth** code from the server that will be exchanged for an authorization token and cached immediately after you've signed in.
+
+1. Click on `Profile` tab to see user information and emails in the page retrieved using the `MSAL.js` library client side.
+
+Did the sample not work for you as expected? Did you encounter issues trying this sample? Then please reach out to us using the [GitHub Issues](../../../../issues) page.
+
+[Consider taking a moment to share your experience with us.](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRz0h_jLR5HNJlvkZAewyoWxUNEFCQ0FSMFlPQTJURkJZMTRZWVJRNkdRMC4u)
+</details>
+
+## About the code
+
+<details>
+ <summary>Expand to see more information about the code</summary>
+
+## Server setup
+
+The entire application is built using the [Microsoft Identity Web](https://docs.microsoft.com/en-us/azure/active-directory/develop/microsoft-identity-web) library and [ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/introduction-to-aspnet-core?view=aspnetcore-6.0) using [Razor](https://docs.microsoft.com/en-us/aspnet/web-pages/overview/getting-started/introducing-razor-syntax-c) pages.
```

Comment:
>?view=aspnetcore-6.0 [](http://example.com/codeflow?start=254&length=20)

if you remove "?view=aspnetcore-6.0" from links, the user will automatically land on the latest version